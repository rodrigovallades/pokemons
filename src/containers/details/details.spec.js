import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { Details } from './index'

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Details container', function () {
  beforeEach(function() {
    global.sessionStorage = jest.genMockFunction();
    global.sessionStorage.setItem = jest.genMockFunction();
    global.sessionStorage.getItem = jest.genMockFunction();
  });

  it('renders "Details" deep correctly', () => {
    const tree = renderer.create(<Details/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders "Details" shallow correctly', () => {
    const component = shallow(<Details/>);
    expect(component.exists()).toEqual(true);
  });

})
