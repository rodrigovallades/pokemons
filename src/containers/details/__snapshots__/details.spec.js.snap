// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details container renders "Details" deep correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div>
      <h1
        className="title"
      >
        <span
          className="float-right"
        >
          <small>
            <Link
              to="/pokemons"
            >
              &lt; pokemons
            </Link>
          </small>
        </span>
         
         
        <small
          className="text-muted"
        >
          details
        </small>
      </h1>
      <div
        className="list-group pokemon"
      >
        <p>
          No details found.
        </p>
      </div>
    </div>
    <div
      className="alert alert-info"
      role="alert"
    >
      <span
        className="octicon octicon-info"
      />
       Tip: Download the 
      <strong>
        <a
          href="https://github.com/zalmoxisus/redux-devtools-extension"
          target="_blnk"
        >
          Redux DevTools
        </a>
      </strong>
       to inspect the Redux store state.
    </div>
  </div>
</div>
`;
